### PR TITLE
Add example for torch.serialization.add_safe_globals

### PR DESCRIPTION
**Pull request by mikaylagawarecki**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #129509
* __->__ #129396
* #129239

### CODE REVIEW

File: torch/serialization.py [hunk 1]
```diff
@@ -207,6 +207,19 @@ def add_safe_globals(safe_globals: List[Any]) -> None:
 
     Args:
         safe_globals (List[Any]): list of globals to mark as safe
+
+    Example:
+        >>> # xdoctest: +SKIP("makes cwd dirty")
```
Comment (albanD) [inline torch/serialization.py:212]: We should use with tempfile...

File: torch/serialization.py [hunk 2]
```diff
@@ -207,6 +207,19 @@ def add_safe_globals(safe_globals: List[Any]) -> None:
...
+        >>> class MyTensor(torch.Tensor):
+        ...     pass
+        >>> t = MyTensor(torch.randn(2, 3))
+        >>> torch.save(t, 'my_tensor.pt')
+        # Running `torch.load('my_tensor.pt', weights_only=True)` will fail with
+        # Unsupported global: GLOBAL __main__.MyTensor was not an allowed global by default.
+        >>> torch.serialization.add_safe_globals([MyTensor])
```
Comment (albanD) [inline torch/serialization.py:219]:
```
# Check the code and making sure it is safe to be used when loaded from an arbitrary checkpoint.
```

**Review by albanD — APPROVED** [review]
nit but SGTM

**Review by malfet — APPROVED** [review]

**Pull request closed without merging**